marpa/lib/Guacamole.pm

```diff
@@ -240,19 +268,32 @@ InterpolString ::= DoubleQuote NonDoubleOrEscapedQuote_Many DoubleQuote
                  | DoubleQuote DoubleQuote
 
 ArrowRHS ::= ArrowDerefCall
+           | ArrowDerefVariable
            | ArrowMethodCall
            | ArrowIndirectCall
            | ArrayElem
            | HashElem
 
-ArrowDerefCall    ::= CallArgs
-ArrowMethodCall   ::= Ident CallArgs
-ArrowIndirectCall ::= SigilScalar Ident CallArgs
+ArrowDerefCall     ::= CallArgs
+ArrowDerefVariable ::= DerefVariableArgsAll
+                     | DerefVariableSlice
+ArrowMethodCall    ::= Ident CallArgs
+ArrowIndirectCall  ::= SigilScalar Ident CallArgs
+
+DerefVariableArgsAll ::= SigilScalar   SigilDerefAll
+                       | SigilArray    SigilDerefAll
+                       | SigilHash     SigilDerefAll
+                       | SigilCode     SigilDerefAll
+                       | SigilGlob     SigilDerefAll
+                       | SigilArrayTop
+
+DerefVariableSlice ::= SigilArray LitArray
+                     | SigilArray LitHash
+                     | SigilHash  LitArray
+                     | SigilHash  LitHash
```

Comment:
To avoid problem with whitespace in `->@ [`, you could explicitly write it out:
```
SigilArrayDeref ~ SigilArray LBracket
PostfixArraySlice ::= SigilArrayDeref Expression RBracket
```

And do the same for `All` variant

Reply:
I'll take a look into that.

Reply:
Okay, this didn't work. Lexemes allow spaces between them. Even when I tried with `'@' '['` it allowed spaces between them. Feature, not bug, I imagine.

Fixed it with `'@[' Expression ...`.